components/transactions/TransactionItem.js

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ChevronDown } from '@styled-icons/feather/ChevronDown';
+import { ChevronUp } from '@styled-icons/feather/ChevronUp';
+import { MessageSquare } from '@styled-icons/feather/MessageSquare';
+import { truncate } from 'lodash';
+import { FormattedDate, FormattedMessage } from 'react-intl';
+
+import { TransactionTypes } from '../../lib/constants/transactions';
+import { formatCurrency } from '../../lib/currency-utils';
+
+import Avatar from '../Avatar';
+import { CreditItem, DebitItem } from '../budget/DebitCreditList';
+import Container from '../Container';
+import DefinedTerm, { Terms } from '../DefinedTerm';
+import ExpenseStatusTag from '../expenses/ExpenseStatusTag';
+import ExpenseTags from '../expenses/ExpenseTags';
+import { Box, Flex } from '../Grid';
+import LinkCollective from '../LinkCollective';
+import OrderStatusTag from '../OrderStatusTag';
+import StyledButton from '../StyledButton';
+import StyledLink from '../StyledLink';
+import { P, Span } from '../Text';
+import TransactionSign from '../TransactionSign';
+import { useUser } from '../UserProvider';
+
+import TransactionDetails from './TransactionDetails';
+import TransactionModal from './TransactionModal';
+
+/** To separate individual information below description */
+const INFO_SEPARATOR = ' • ';
+
+const TransactionItem = transaction => {
+  const {
+    toAccount,
+    fromAccount,
+    order,
+    expense,
+    type,
+    amount,
+    netAmount,
+    description,
+    usingVirtualCardFromCollective,
+    createdAt,
+    isRefunded,
+  } = transaction;
+  const { LoggedInUser } = useUser();
+  const [isExpanded, setExpanded] = React.useState(false);
+  const isOrder = order !== null;
+  const isCredit = type === TransactionTypes.CREDIT;
+  const Item = isCredit ? CreditItem : DebitItem;
+
+  const avatarCollective = isOrder ? (isCredit ? fromAccount : toAccount) : isCredit ? fromAccount : toAccount;
+
+  const isRoot = LoggedInUser && LoggedInUser.isRoot();
+  const isHostAdmin = LoggedInUser && LoggedInUser.isHostAdmin(fromAccount);
+  const isFromCollectiveAdmin = LoggedInUser && LoggedInUser.canEditCollective(fromAccount);
+  const isToCollectiveAdmin = LoggedInUser && LoggedInUser.canEditCollective(toAccount);
+  const canDownloadInvoice = isRoot || isHostAdmin || isFromCollectiveAdmin || isToCollectiveAdmin;
```

Comment:
(Can be done later) Would be great to introduce a `permissions` field similar to [the on on Expense](https://github.com/opencollective/opencollective-api/blob/dad2d60b946f1d1dbbd7eddc6605647593560205/server/graphql/v2/object/Expense.js#L187) so that we don't have to do all these checks frontend-side:

```gql
query {
  transaction {
    permissions {
      canDownloadInvoice
      canRefund
    }
  }
}
```